### PR TITLE
UI: Use server-sent error messages when applicable

### LINKED ISSUE #7875: "allocation lifecycle permissions" when trying to restart with a stopped allocation
### Nomad version
Nomad v0.11.1 (b43457070037800fcc8442c8ff095ff4005dab33)

### Operating system and Environment details
Linux X 5.4.0-28-generic #32-Ubuntu SMP Wed Apr 22 17:40:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
Docker Engine - Community
 Version:           19.03.8
Ubuntu 20.04

### Issue
When trying to restart my allocation or my job I get the following error.
`Your ACL token does not grant allocation lifecycle permissions.`
even when using a management token. 

When I click restart I see this in my logs:
```
 2020-05-05T23:12:20.126-0500 [ERROR] http: request failed: method=PUT path=/v1/client/allocation/31b5957a-7893-5766-83e8-f697e41b6d7d/restart error="Task not running" code=500
```
So this seem to be caused when an allocation fails to start or is otherwise not running and you try to restart.

A quick search for the error in the code suggests it gets shown for any exception which tracks with the token having permissions and the error in the log so I'm under the assumption that's the cause.

### Reproduction steps
I think you can reproduce this by starting a job, killing the allocation till its stops respawning and trying to restart. I haven't nailed down the steps yet.

### Nomad Client logs (if appropriate)
Above

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
Fixes #7875 

In most places in the UI, the error message shown after failed actions is a mostly generic "Your ACL token said no" message. Sometimes this occludes genuine and helpful 500 error messages being returned from the API.

This makes it so anywhere we show an error message we make sure to only show an ACL error message if it is a 403 error, otherwise just print what the API gave us.

### CODE REVIEW

File: ui/app/controllers/allocations/allocation/index.js [hunk 1]
```diff
@@ -85,7 +86,7 @@ export default class IndexController extends Controller.extend(Sortable) {
     } catch (err) {
       this.set('error', {
         title: 'Could Not Restart Allocation',
-        description: 'Your ACL token does not grant allocation lifecycle permissions.',
+        description: messageForError(err, 'manage allocation life cycle'),
```
Comment (backspace) [inline ui/app/controllers/allocations/allocation/index.js:89]: The [documentation](https://www.nomadproject.io/docs/job-specification/lifecycle) seems to consistently use `lifecycle` vs `life cycle`, should this too?

**Review by backspace — CHANGES_REQUESTED** [review]
I appreciate this extraction. I wonder whether it might be improved by having a test that exercises the `error.errors` branch when it’s not a 403?

File: ui/app/controllers/allocations/allocation/index.js (hunk 1 above)
Comment (DingoEatingFuzz) [reply ui/app/controllers/allocations/allocation/index.js:89]: Good call.

**Comment by DingoEatingFuzz** [comment]
Okay! I reverted the `life cycle` change and added unit tests. Ready for re-review ✨

**Review by backspace — APPROVED** [review]
I should have been more specific in that I was imagining an acceptance test so it’s more end-to-end-y but really I think this is fine, thanks 🥳

**Comment by DingoEatingFuzz** [comment]
Oh! I like that idea. It also doubles as evidence in the UI code that we expect plain text responses from errors.

**Pull request merged**